src/mercat/transaction.rs

```diff
@@ -920,7 +922,6 @@ mod tests {
         let ctx_init_data = result.unwrap();
 
         // Finalize the transaction and check its state
-        let tx_id = tx_id + 1;
```

Comment:
We could have a negative test here to check for the `TransactionIdMismatch` error.

Reply:
Sure, will add that.